binding/jms/src/test/java/io/tracee/binding/jms/TraceeMessageListenerTest.java

```diff
@@ -47,22 +47,31 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
 
 	@Test
 	public void testDecodesFromMessage() throws Exception {
-		encodedContext.put("contextFromMessage","yes");
+		encodedContext.put(INVOCATION_ID_KEY, "yes");
 		unit.intercept(invocationContext);
-		assertThat(backend.getValuesBeforeLastClear(), hasEntry("contextFromMessage", "yes"));
+		assertThat(backend.getValuesBeforeLastClear(), hasEntry(INVOCATION_ID_KEY, "yes"));
+		verify(mdbLike).onMessage(message);
 	}
 
 	@Test
 	public void testCleansUpContextAfterProcessing() throws Exception {
-		encodedContext.put("contextFromMessage","yes");
+		encodedContext.put("contextFromMessage", "yes");
 		unit.intercept(invocationContext);
 		assertThat(backend.isEmpty(), is(true));
+		verify(mdbLike).onMessage(message);
+	}
+
+	@Test
+	public void testInvocationIdGeneratedIfAbsent() throws Exception {
+		unit.intercept(invocationContext);
```

Comment:
Hmm in general I like focused PRs. For example if the focus is a bugfix, only the code necessary for the bugfix should be inside - nothing else. If you're focused on some refactoring / code cleaning, only this should be inside of the PR.
With your bugfix you should just change the entire test cases / add new ones to test your new behavior.
Regarding the deletion of the line / assert: We check at this point that the `UNIT` ( `TraceeMessageListener`) is doing his job by reading the TPIC from the message into the backend. This is a bit strange in TracEE because most of the test cases simulate the sender and the receiver to check the correct behavior. 

Regarding the mdcLike variable: I'm fine with the variable :-)